### PR TITLE
feat(gov): vote receipts — persist, reconcile, short-circuit relays (PR 6a)

### PR DESCRIPTION
**Pull request by sidhujag**
## Summary
PR 6a — **correctness foundation** for the Governance Vote Intelligence arc. After this PR the backend persistently remembers which vault-owned masternodes voted on which proposal, reconciles against the chain, and uses that knowledge to keep votes correct and retries smart — **without changing the wire protocol of `POST /gov/vote`**.

### Data model
- New `vote_receipts` table keyed on `(user_id, proposal_hash, collateral_hash, collateral_index)` with indexes for per-user listings and per-proposal reconciliation (`db/migrations/001_init.sql`).

### `lib/voteReceipts.js`
- Parses Core's `gobject_getcurrentvotes` output (same format Core emits via `CGovernanceVote::ToString`: `<txid>-<n>:<nTime>:<outcome>:<signal>`).
- Repo: `upsert / listForProposal / summaryForUser / decideRelay / reconcileForProposal`.
- `reconcileForProposal()` marks receipts **confirmed** when the chain agrees, **stale** when the chain disagrees (cross-device vote change), and leaves them **relayed** otherwise.
- `createCurrentVotesCache`: per-process LRU-of-one-per-proposal with short TTL to coalesce concurrent callers onto a single `gobject_getcurrentvotes` RPC.

### Smart relay in `lib/gov.js`
- `relayVotes` consults `decideRelay` before each `voteraw`. Already-confirmed and recently-relayed entries **short-circuit** (returning `ok:true` with a `skipped` reason) so duplicate attempts don't burn a `vote_too_often` rate-limit hit.
- Outcomes are persisted on success **and** failure so the frontend can distinguish 'never tried' from 'tried and failed' for retry UX.

### New endpoints (`routes/gov.js`)
- `GET /gov/receipts?proposalHash=<hex64>&refresh=0|1` — user's stored receipts for a proposal, optionally reconciled against the chain. Skips the RPC when all receipts are confirmed and `verified_at` is within the freshness window (unless `refresh=1`). RPC failures surface as `reconcileError` in the body without blocking.
- `GET /gov/receipts/summary` — per-proposal rollup of status counts for the user. Both require auth; CSRF via existing double-submit middleware.

### Wiring
- `server.js` creates the `currentVotesCache` and threads `getCurrentVotes` through `mountAuthAndVault` / `createGovRouter`. `lib/appFactory.js` integrates `createVoteReceiptsRepo` into `buildServices`.

## Test plan
- [x] `lib/voteReceipts.test.js` — golden `CGovernanceVote::ToString` fixture, parser edge cases, cache concurrency, upsert / `decideRelay` / `reconcileForProposal` scenarios.
- [x] `lib/gov.test.js` — strict `voteSig` decode + `relayVotes` paths (already-on-chain / recently-relayed / persisted-failure).
- [x] `tests/gov.routes.test.js` — receipt persistence via `POST /gov/vote`, short-circuit behaviour, freshness-window cache skip, `refresh=1` forcing reconciliation, graceful RPC-failure handling, user isolation, full `GET /gov/receipts` + `/gov/receipts/summary` coverage.
- [x] Full suite: **352/352 passing**.

## Out of scope (deferred)
- Per-row live-status transitions, exhaustive error-copy branches, cohort-aware chips, ops summary / activity card → **PR 6b / 6c**.

---
Paired frontend PR: syscoin/sysnode-info#gov-vote-receipts

Made with [Cursor](https://cursor.com)

### CODE REVIEW

File: db/migrations/001_init.sql [hunk 1]
```diff
@@ -129,3 +141,32 @@ CREATE INDEX idx_pending_registrations_email
...
+  proposal_hash    TEXT    NOT NULL,
+  vote_outcome     TEXT    NOT NULL,
+  vote_signal      TEXT    NOT NULL,
+  vote_time        INTEGER NOT NULL,
+  status           TEXT    NOT NULL,
+  last_error       TEXT,
+  submitted_at     INTEGER NOT NULL,
+  verified_at      INTEGER,
+  UNIQUE(user_id, collateral_txid, collateral_vout, proposal_hash)
+);
```
[the remark this answers is not in the document]
Comment (sidhujag) [reply db/migrations/001_init.sql:167]: N/A

[the remark this answers is not in the document]
Comment (sidhujag) [reply db/migrations/001_init.sql:167]: Dismissing this one: there is no deployed database anywhere (no prod, no staging, no shared dev instance) that has `001_init.sql` recorded in `schema_migrations`, and the repo is pre-first-deploy. Every install goes through `001_init.sql` fresh so appending the DDL has the same effect as a new migration file.

I'll honor the immutable-migrations invariant the moment there's a durable DB in flight (PR 6b/6c territory at the latest). Flagging this as an assumption so future contributors can sanity-check it.

**Comment by sidhujag** [comment]
@codex review

File: db/migrations/001_init.sql [hunk 2]
```diff
@@ -129,3 +141,32 @@ CREATE INDEX idx_pending_registrations_email
...
+
+-- vote_receipts: persistent record of each governance vote we relayed on
+-- behalf of a user, scoped per (user, MN, proposal). Populated by
+-- /gov/vote on both successful and failed `voteraw` calls so the UI can
+-- distinguish "already relayed" from "needs retry", and the on-demand
+-- reconciler can flip rows to 'confirmed' or 'stale' after comparing
+-- against Core's gobject_getcurrentvotes. UNIQUE on (user, outpoint,
+-- proposal) makes a vote change an UPDATE in place rather than a new
+-- row; receipts track current intent, not history.
+CREATE TABLE vote_receipts (
```
[the remark this answers is not in the document]
Comment (sidhujag) [reply db/migrations/001_init.sql:153]: N/A

**Comment by sidhujag** [comment]
@codex review

**Comment by sidhujag** [comment]
@codex review

Context for this round:
- The `vote_receipts` DDL in `db/migrations/001_init.sql` is intentional and NOT a bug. There is no deployed database anywhere (no prod, no staging, no shared dev instance); every install runs `001_init.sql` fresh, so appending the DDL is equivalent to a new migration file. Please skip the "add forward migration" comment on this PR. We'll adopt the immutable-migrations rule the moment there's a durable DB in flight.
- Please re-review the latest commit (`ffd0bce`) which adds a freshness check for the `confirmed` skip path in `decideRelay`, and the prior `7e72534` which evicts expired entries from `createCurrentVotesCache`.

**Comment by sidhujag** [comment]
@codex review

Reminder for this pass: the `vote_receipts` DDL in `001_init.sql` is intentional (no deployed DB exists — see earlier reply). Please skip the "add forward migration" comment.

**Comment by sidhujag** [comment]
@codex review

Reminder: the `vote_receipts` DDL in `001_init.sql` is intentional (no deployed DB anywhere — every install runs it fresh). Please skip the "add forward migration" comment.

**Comment by sidhujag** [comment]
@codex review

Reminder: the `vote_receipts` DDL in `001_init.sql` is intentional (no deployed DB anywhere). Please skip that comment.

**Comment by sidhujag** [comment]
@codex review

Reminder: the vote_receipts DDL in 001_init.sql is intentional (no deployed DB anywhere). Please skip that comment. partner PR https://github.com/syscoin/sysnode-info/pull/7

**Comment by sidhujag** [comment]
@codex review Reminder: the vote_receipts DDL in 001_init.sql is intentional (no deployed DB anywhere — every install runs it fresh). Please skip the "add forward migration" comment.

Commit 1863529 addresses round-6 P1: cold-start duplicate-vote responses now persist an anchor receipt (status:'relayed', lastError:null) so reconcileForProposal has a row to backfill against. A prior 'confirmed' or 'relayed' receipt is still preserved (not clobbered by 'failed'); prior 'failed'/'stale' is upgraded to 'relayed' on a duplicate-vote. 373/373 backend tests pass.

**Comment by sidhujag** [comment]
@codex review Reminder: the vote_receipts DDL in 001_init.sql is intentional (no deployed DB anywhere — every install runs it fresh). Please skip the "add forward migration" comment.

Commit e40f393 addresses the round-7 P2 on createCurrentVotesCache: in-flight entries now carry expiresAt=Infinity (so a slow RPC doesn't age out mid-flight and amplify load) and TTL is stamped from resolution time (so a slow RPC still yields a full ttlMs post-resolution fresh window). 375/375 backend tests pass.
